src/perform_step/ssprk_perform_step.jl

```diff
@@ -1119,4 +1119,87 @@ end
 
   f( k,  u, p, t+dt)
 end
-  
\ No newline at end of file
+
+
+function initialize!(integrator,cache::RKMConstantCache)
+	integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+	integrator.kshortsize = 2
+	integrator.k = typeof(integrator.k)(undef, integrator.kshortsize)
+  
+	# Avoid undefined entries if k is an array of arrays
+	integrator.fsallast = zero(integrator.fsalfirst)
+	integrator.k[1] = integrator.fsalfirst
+	integrator.k[2] = integrator.fsallast
+end
+
+@muladd function perform_step!(integrator,cache::RKMConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack α2,α3,α4,α5,α6,β1,β2,β3,β4,β5,β6,c2,c3,c4,c5,c6 = cache
+
+  # u1
+  k1 =integrator.fsalfirst
+  u   = uprev + β1*dt*k1
+  # u2
+  k2 = f(uprev + α2*k1*dt, p, t+c2*dt)
+  u  = u + β2*dt*k2
+  # u3
+  k3 = f(uprev + α3*k2*dt, p, t+c3*dt)
+  u   = u + β3*dt*k3
+  # u4
+  k4 = dt*f(uprev + α4*k3*dt, p, t+c4*dt)
```

Comment:
don't do `dt*f` if the `dt` is handled later!

Reply:
I am aware of rateType and uType. Actually i changed the implementation, firstly i was using tmp (uType) then i implemented using k1,k2.. instead. I should have checked carefully.
Sorry for the inconvenience caused.